### PR TITLE
X-Function-Status callback request header documented

### PR DESCRIPTION
**Pull request by ewilde**
Signed-off-by: Edward Wilde <ewilde@gmail.com>

### CODE REVIEW

File: docs/deployment/troubleshooting.md [hunk 1]
```diff
@@ -25,6 +25,15 @@ The [OpenFaaS workshop](https://github.com/openfaas/workshop) has more instructi
 
 The Queue Worker component enables asynchronous processing of function requests. The default verbosity level hides the message content, but this can be viewed by setting write_debug to true when deploying.
 
+* Callback request headers
+
+The following additional request headers will be set when invoking the call back url:
+
+| Header            | Description |
+|-------------------|-------------|
+| X-Function-Status | [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) returned by the original function call |
```
Comment (rgee0) [inline docs/deployment/troubleshooting.md]: To offset future queries I wonder if the table should cover the full set:
```
X-Call-Id
X-Duration-Seconds
X-Function-Status
X-Request-Id
```

File: docs/deployment/troubleshooting.md [hunk 2]
```diff
@@ -25,6 +25,15 @@ The [OpenFaaS workshop](https://github.com/openfaas/workshop) has more instructi
 
 The Queue Worker component enables asynchronous processing of function requests. The default verbosity level hides the message content, but this can be viewed by setting write_debug to true when deploying.
 
+* Callback request headers
+
+The following additional request headers will be set when invoking the call back url:
```
Comment (rgee0) [inline docs/deployment/troubleshooting.md]: url => URL

File: docs/deployment/troubleshooting.md (hunk 1 above)
Comment (alexellis) [reply docs/deployment/troubleshooting.md]: What is X-Request-Id?

Comment (rgee0) [reply docs/deployment/troubleshooting.md]: C&P remnant

Comment (ewilde) [reply docs/deployment/troubleshooting.md]: @rgee0 sound like a good suggestion, i've pushed a new commit with those other two headers

File: docs/deployment/troubleshooting.md (hunk 2 above)
Comment (ewilde) [reply docs/deployment/troubleshooting.md]: fixed

File: docs/deployment/troubleshooting.md [hunk 3]
```diff
@@ -25,6 +25,17 @@ The [OpenFaaS workshop](https://github.com/openfaas/workshop) has more instructi
 
 The Queue Worker component enables asynchronous processing of function requests. The default verbosity level hides the message content, but this can be viewed by setting write_debug to true when deploying.
 
+### Callback request headers
```
Comment (ewilde) [inline docs/deployment/troubleshooting.md]: made this a heading so I can link to it from the workshop on async functions

**Comment by alexellis** [comment]
I like it :+1:

**Pull request merged**